corehq/motech/forms.py

```diff
@@ -62,81 +66,70 @@ def __init__(self, *args, domain, **kwargs):
                 kwargs['initial'] = {'plaintext_password': PASSWORD_PLACEHOLDER}
         super().__init__(*args, **kwargs)
 
-    def save(self, commit=True):
-        self.instance.domain = self.domain
-        self.instance.plaintext_password = self.cleaned_data['plaintext_password']
-        return super().save(commit)
-
-
-class BaseConnectionSettingsFormSet(forms.BaseModelFormSet):
-
-    def __init__(self, *args, domain, **kwargs):
-        super().__init__(*args, **kwargs)
         self.domain = domain
-        self.form_kwargs['domain'] = domain  # Passed by ``FormView.get_form_kwargs()``
-
-    def clean(self):
-        super().clean()
-        errors = [f'Unable to delete connection "{f.instance}": It is in use.'
-                  for f in self.not_deleted_forms]
-        if errors:
-            raise ValidationError(errors)
-
-    @property
-    def not_deleted_forms(self):
-        """
-        Returns a list of forms marked for deletion but whose
-        ConnectionSettings are in use.
-        """
-        deleted_forms = super().deleted_forms
-        ids_in_use = get_connection_ids_in_use(self.domain)
-        return [f for f in deleted_forms
-                if f.is_bound and f.instance.id in ids_in_use]
-
-
-def get_connection_ids_in_use(domain):
-    from corehq.motech.dhis2.dbaccessors import get_dataset_maps
-
-    dataset_maps = get_dataset_maps(domain)
-    # So far only DataSetMaps use ConnectionSettings. When more things
-    # do (like Repeaters), this must check those too.
-    return {m.connection_settings_id for m in dataset_maps
-            if m.connection_settings_id}
-
-
-ConnectionSettingsFormSet = forms.modelformset_factory(
-    model=ConnectionSettings,
-    form=ConnectionSettingsForm,
-    formset=BaseConnectionSettingsFormSet,
-    extra=1,
-    can_delete=True,
-)
-
-
-class ConnectionSettingsFormSetHelper(FormHelper):
-
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.form_class = 'form-horizontal'
-        self.label_class = 'col-sm-3 col-md-2'
-        self.field_class = 'col-sm-9 col-md-8 col-lg-6'
-        self.layout = crispy.Layout(
+        self.helper = hqcrispy.HQFormHelper()
+        self.helper.form_class = 'form-horizontal'
```

Comment:
Using `HQFormHelper` means you don't need `form_class`, `label_class`, or `field_class`.

Reply:
Thank you. 0815a541cd7